### ISSUE TITLE
This year's sessions have last year's rooms & times

### ISSUE THREAD

**Issue by ForestMars** [NYC-Camp/website#170]
All this year's session nodes are still displaying (in the admin view) last year's Session Time and Room Assignment fields and do not have any fields for this year's timeslot or room assignment. 

Do we already have a model of how we are assigning timeslot & room assignment fields to conferences each year? 

Keep in mind we can't simply delete those fields, since that would mean previous year's sessions would loose that information, which breaks the data contact ("Drupal will never destroy your data.")

Obviously this is integral and needed for the session grid, is it there and I am just missing it, or is it in a feature branch on multi-dev?

**Comment by likewhoa** [comment]
Where is the admin view?

**Comment by ForestMars** [comment]
By admin view I meant node/edit

**Comment by likewhoa** [comment]
looks like we'll need a contextual filter on these entity reference fields in order to show data relevant to the current year for both fields.

**Comment by likewhoa** [comment]
I'll look into this today to see what will work, this should be part of the COD roadmap. Do we have a link to this part of the roadmap upstream?

**Comment by likewhoa** [comment]
The entityreference view in question for timeslots is located at admin/structure/views/view/cod_session_timeslot/edit/entityreference_1
for room assignments we are not using an entityreference field so it's basically showing all room content types. This should be converted into an entityreference field to filter by year.

**Comment by ForestMars** [comment]
The COD Roadmap page is pretty general: http://usecod.com/roadmap

Most of the more specific discussion is taking place in the COD Support modules issue queue.

**Comment by likewhoa** [comment]
do you know who introduced the room assignment? it seems upstream doesn't have this field in session content type.

![untitled](https://f.cloud.github.com/assets/1696920/2441340/1a7c2c00-ae13-11e3-8be9-3e9df24d228a.png)

**Comment by likewhoa** [comment]
I when ahead an made a few changes but display data for last years room assignment is lost.. The following changes and workaround have been added.
- Added a new view to reference rooms from 2014 only
- Changed 'field_room_assignment' from a simple entity selection option to a views entity reference mode
- Republished the old room assignment nodes since these won't show up in session edit view now

We really need to figure out a better way to filter old camp data in cod since now it's all in one place and cod didn't really think about scaling in the sense that camps might have different locations every year.

**Comment by ForestMars** [comment]
Hopefully will be able to test this in a bit, working on the schedule now...

**Comment by esod** [comment]
Changing the field_room_assignment from a simple entity selection option to a views entity reference probably breaks My Schedule. I should be able to adjust the view to suit.

Also, @likewhoa your field list is not up to date. Have a look at this screen shot from live.

![screen shot 2014-03-19 at 8 57 24 am](https://f.cloud.github.com/assets/387576/2460136/768bb138-af66-11e3-98ab-df6888322eeb.png)

I'll default the features from live.

**Comment by esod** [comment]
Sorry @likewhoa reread your post. Your screen shot was from upstream. I've been calling it cod out of the box (oob).

Yes, we added field_room_assignment.

**Comment by esod** [comment]
Changing field_room_assignment doesn't affect the My Schedules tab. But something has changed in the Display Suite (ds) that's preventing the Add to my Schedule/Remove from my schedule button from printing. I see that session content type doesn't have a ds layout. So even if I create a custom ds field, it won't appear in /admin/structure/types/manage/session/display. If I add a ds layout to the session content type the view of the session nodes goes haywire. I tested this out only on dev and test. live is untouched.

I can work around this by adding the session_schedule flag in from the flag admin. I'd prefer that the template code I wrote were not being intercepted by display suite. The code prints the Add to my Schedule/Remove from my schedule button only on Accepted sessions.

**Comment by ForestMars** [comment]
Do you think we should do the session schedule grid first, and then the add to my schedule? The former will be more widely used (not everyone uses add to my schedule, but everyone needs to be able to see the schedule.) 

It should really just be updating the view from last year, with new times and rooms, I think...?

**Comment by esod** [comment]
Aren't the session schedule views a part of NYC Camp Schedule view? 

I'm really not liking Display Suite because we've only implemented it halfway by not using layouts or its text format. I tried to add a layout to the user ds on my local so I can get at #165 and the user page loses all its formatting. Grrr...

**Comment by ForestMars** [comment]
Yes, although last year I just added the view but didn't featurize it, which we should do. Unlike previous years, this year we won't be having other sessions on Sunday, so Saturday is the main day that we'll be using the room x time grid for. 

If we're serious about making our improvements available  to other camps, we should probably create master displays, and consider using the approach to features overrides we talked about. Any work we do on the schedule view kind of straddles the line between "things to do post-camp to contribute back" and "decoupling our specific use case so our work is useful to others" if that makes sense. 

My recollection is that the session grid that shipped with COD alpha2 was completely unusable (I don't think it was even claimed to work) so we should see where that's at in alpha 5, given that we didn't contribute our changes back last year, unless they were added to the Codify module.

**Comment by ForestMars** [comment]
I feel like this issue is blowing up a bit to include the session grid view, which was not the original scope. 

Currently, all sessions are still showing last year's choices for time slots. This is probably due to the ppoor implementation method, hard wiring a time slot (eg. "10:00am-10:45am) to a fixed date. That makes no sense whatsoever to me. It seems obvious that time slot should be a reusable field every year, and then each year we add to that the day/date a session is scheduled. 

This is a critically important issue now that we are less than 3 weeks away from the camp.

**Comment by ForestMars** [comment]
Where are we at on this? The session selection team would like to finalize the schedule, and needs to be able to view the grid to do this.

**Comment by willykaram** [comment]
Rather than apply a band aid solution here, I've been working to fix the issue properly, and in keeping with the architecture problem that forest noted above. I'll be done by midday today. Will post a summary here shortly for review.

**Comment by likewhoa** [comment]
@willykaram Can you post the summary of any changes you made to this for the record, otherwise lets close this issue.

**Comment by willykaram** [comment]
@likewhoa This is a summary of the the strategy that was implemented based on some of my discussions with Forest:

https://github.com/NYC-Camp/website/wiki/NYC-Camp-Conference-Scheduling

It's what's been used to the the scheduling for the sessions/keynotes so far. As by today, we should have it implemented to the extent needed for any of the summits/trainings which will resolve issue #205 and #249 .

Although it's going to take more planning/work to resolve some of the multi-year issues like #170 , which we can address after the camp (ideally just with some expose filters on UIs and contextual filters on views. Part of the rational for using the chosen module is it give us a better UI for work around the problem mentioned in this issue (still nowhere near a solution, just less painful for now).

**Comment by likewhoa** [comment]
@willykaram not sure an expose filter will be ideal in this case since we want to archive old conferences i.e (2013.nyccamp.org) and over years it would be a PITA to have a large year drop down, so instead there should be a contextual filter based on the current year (when node was created) so that only relevant content shows up.
